File: 01.two-sum/two_sum.go
package leetcode

func TwoSum(nums []int, target int) []int {
	tempMap := map[int]int{}
	for n, i := range nums {
		m, ok := tempMap[target-i]
		if ok {
			return []int{m, n}
		}
		tempMap[i] = n
	}
	return []int{}
}

File: 01.two-sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        n = 0
        for i in nums:
            if (target - i) in m:
                return m[target - i], n
            m[i] = n
            n = n + 1

